http-common/Cargo.toml

```diff
@@ -20,7 +20,8 @@ openssl = { version = "0.10", optional = true }
 percent-encoding = "2"
 serde = { version = "1", features = ["derive"] }
 serde_json = "1"
-tokio = { version = "1", features = ["net", "rt-multi-thread"], optional = true }
+tokio = { version = "1", features = ["net", "rt-multi-thread", "sync"], optional = true }
+tokio-io-timeout = { version = "1" }
```

Comment:
`tokio-io-timeout` by itself is not enough. That sets the limit for every individual read, not for the whole read of the request. That means the client can drip-feed one byte every 5s and the request won't time out.

You also need to apply a [`tokio::time::Timeout`](https://docs.rs/tokio/1.5.0/tokio/time/fn.timeout.html) on all the calls to `hyper::body::to_bytes(body)` inside the `make_service` macro.

Reply:
Fixed